e2e/runJest.ts

```diff
@@ -208,17 +235,25 @@ export const runContinuous = function (
     },
 
     async waitUntil(fn: ConditionFunction) {
-      await new Promise<void>(resolve => {
-        const check = (state: StdErrAndOutString) => {
+      await new Promise<void>((resolve, reject) => {
+        const check: CheckerFunction = state => {
           if (fn(state)) {
             pending.delete(check);
+            pendingRejection.delete(check);
             resolve();
           }
         };
+        const error = new ErrorWithStack(
+          'Process exited',
+          continuousRun.waitUntil,
+        );
```

Comment:
create the error here to point back to the `waitUntil`

![image](https://user-images.githubusercontent.com/1404810/135268694-76217bac-6363-46fd-80f6-2bf0e9a156e1.png)